src/analog/private/m2kanalogout_impl.cpp

```diff
@@ -264,7 +264,7 @@ class M2kAnalogOut::M2kAnalogOutImpl : public DeviceGeneric {
 		m_samplerate.at(1) = getSampleRate(1);
 
 		for (unsigned int i = 0; i < nb_samples; i++) {
```

Comment:
unrelated to this PR; and looking at this in a more general-mode, is there a way to do these without doing the for-loop?
this looks very C-ish, and looks like it could benefit from more C++ constructs;

would something like this work?
```
std::vector<short> raw_data_buffer(data, data + nb_samples);
```

Reply:
This makes sense in some cases. Let me add some changes and check more thoroughly for these in the future.